packages/common/src/util.js

```diff
@@ -837,3 +833,42 @@ function determineBaseUrl(options, environmentVariables,
 }
 
 util.determineBaseUrl = determineBaseUrl;
+
+/**
+ * Resolves the api endpoint to use, the port and whether API
+ * calls need to be made with the credentials passed in the args.
+ *
+ * @param {object=} options - Service configuration object.
+ * @param {array} environmentVariables - Array of accepted environment
+ *     variables through which a custom api endpoint can be specified.
+ * @param {string=} defaultApiEndpoint - Fallback api endpoint to use.
+ * @param {number} defaultPort - Fallback port to use.
+ * @param {object=} credentials - An instance of grpc.ServerCredentials
+ *     these credentials will be used if the api endpoint is considered
+ *     to be a custom endpoint
+ * */
+function resolveGapicOptions(options, environmentVariables,
```

Comment:
Would you switch this method to accept an object instead of multiple arguments?

Reply:
Although generally I'm against functions with many arguments (this one has 5, it's borderline too many) in this case I kind of like the fact that it's consistent with determineBaseUrl in the sense that it's very obvious from the arguments that these two functions are somehow related.

When I read index.js in a "common" service and a "gapic" based service, the custom endpoint bits look similar and it's easy to follow what's going on. If we put everything under an object, not only it's less explicit but I'll have to unmarshall the object anyway to pass the args down to determineUrl

Other than that if you feel like 5 args is too many or maybe have another reason that I'm missing (new lint rules for example?) I can make the changes